src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -682,12 +732,29 @@ export default class DataTable extends Vue {
     if (selectAllCheckbox) {
       const isSelected = (row: DataTableRow) => this.selectedRows.includes(row.rowId);
 
-      if (this.slicedData) {
-        selectAllCheckbox.checked = this.slicedData.every(isSelected);
+      if (this.slicedData?.every(isSelected)) {
+        this._changeCheckboxState(selectAllCheckbox, true, false);
+      } else if (this.slicedData?.some(isSelected)) {
+        this._changeCheckboxState(selectAllCheckbox, false, true);
+      } else {
+        this._changeCheckboxState(selectAllCheckbox, false, false);
       }
     }
   }
 
+  _changeCheckboxState(check: HTMLInputElement, checked: boolean, indeterminate: boolean) {
+    check.checked = checked;
+    check.indeterminate = indeterminate;
+
+    if (checked) {
+      check.classList.remove(INDETERMINATE_CLASS);
+    } else if (!checked && !indeterminate) {
```

Comment:
```suggestion
    if (checked || !indeterminate) {
```

Reply:
I changed the condition. now I think it is more clear.

Reply:
I saw it, and I think there is one branch which you don't need. You can simplify it like this:

if (!checked && indeterminate) {
      check.classList.add(INDETERMINATE_CLASS);
    } else {
      check.classList.remove(INDETERMINATE_CLASS);
    }

Reply:
True, thanks!